diff --git a/node_modules/mpv.js/build/Release/mpv-1.dll b/node_modules/mpv.js/build/Release/mpv-1.dll
new file mode 100644
index 0000000..f95ea5e
Binary files /dev/null and b/node_modules/mpv.js/build/Release/mpv-1.dll differ
diff --git a/node_modules/mpv.js/index.js b/node_modules/mpv.js/index.js
index f67ceaa..fc8bfb9 100644
--- a/node_modules/mpv.js/index.js
+++ b/node_modules/mpv.js/index.js
@@ -5,7 +5,6 @@
 
 "use strict";
 
-const path = require("path");
 const React = require("react");
 const PropTypes = require("prop-types");
 
@@ -31,6 +30,7 @@ function containsNonASCII(str) {
  * @throws {Error} Resulting path contains non-ASCII characters.
  */
 function getPluginEntry(pluginDir, pluginName = "mpvjs.node") {
+  const path = require("path");
   const fullPluginPath = path.join(pluginDir, pluginName);
   // Try relative path to workaround ASCII-only path restriction.
   let pluginPath = path.relative(process.cwd(), fullPluginPath);
