diff --git a/node_modules/react-native-vlcplayer2/android/build.gradle b/node_modules/react-native-vlcplayer2/android/build.gradle
index f798a43..2dbbe91 100644
--- a/node_modules/react-native-vlcplayer2/android/build.gradle
+++ b/node_modules/react-native-vlcplayer2/android/build.gradle
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion '25.0.0'
+    compileSdkVersion 28
+    buildToolsVersion '28.0.3'
 
     defaultConfig {
         minSdkVersion 17
-        targetSdkVersion 23
+        targetSdkVersion 28
         versionCode 1
         versionName "1.0"
         ndk {
diff --git a/node_modules/react-native-vlcplayer2/ios/RCTVLCPlayer/RCTVLCPlayer.m b/node_modules/react-native-vlcplayer2/ios/RCTVLCPlayer/RCTVLCPlayer.m
index eaeb530..30720e7 100644
--- a/node_modules/react-native-vlcplayer2/ios/RCTVLCPlayer/RCTVLCPlayer.m
+++ b/node_modules/react-native-vlcplayer2/ios/RCTVLCPlayer/RCTVLCPlayer.m
@@ -239,7 +239,7 @@ static NSString *const playbackRate = @"rate";
         int remainingTime = [[_player remainingTime] intValue];
         int duration      = [_player.media.length intValue];
         
-        if( currentTime >= 0 && currentTime < duration) {
+        if( currentTime >= 0 ) {
             self.onVideoProgress(@{
                                    @"target": self.reactTag,
                                    @"currentTime": [NSNumber numberWithInt:currentTime],
