diff --git a/node_modules/react-native-yz-vlcplayer/ios/RCTVLCPlayer/RCTVLCPlayer.m b/node_modules/react-native-yz-vlcplayer/ios/RCTVLCPlayer/RCTVLCPlayer.m
index 03454ea..86d3917 100644
--- a/node_modules/react-native-yz-vlcplayer/ios/RCTVLCPlayer/RCTVLCPlayer.m
+++ b/node_modules/react-native-yz-vlcplayer/ios/RCTVLCPlayer/RCTVLCPlayer.m
@@ -371,7 +371,7 @@ static NSString *const playbackRate = @"rate";
         int remainingTime = [[_player remainingTime] intValue];
         int duration      = [_player.media.length intValue];

-        if( currentTime >= 0 && currentTime < duration) {
+        if( currentTime >= 0) {
             self.onVideoProgress(@{
                                    @"target": self.reactTag,
                                    @"currentTime": [NSNumber numberWithInt:currentTime],
