diff --git a/node_modules/react-native-svg/ios/Utils/RNSVGPathMeasure.m b/node_modules/react-native-svg/ios/Utils/RNSVGPathMeasure.m
index ed32e1f..416b29d 100644
--- a/node_modules/react-native-svg/ios/Utils/RNSVGPathMeasure.m
+++ b/node_modules/react-native-svg/ios/Utils/RNSVGPathMeasure.m
@@ -20,7 +20,7 @@
 /**
  * returns the distance between two points
  */
-CGFloat distance(CGPoint p1, CGPoint p2)
+static CGFloat distance(CGPoint p1, CGPoint p2)
 {
     CGFloat dx = p2.x - p1.x;
     CGFloat dy = p2.y - p1.y;
