diff --git a/node_modules/react-native-vlcplayer2/android/build.gradle b/node_modules/react-native-vlcplayer2/android/build.gradle
index f798a43..8d7f5cb 100644
--- a/node_modules/react-native-vlcplayer2/android/build.gradle
+++ b/node_modules/react-native-vlcplayer2/android/build.gradle
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion '25.0.0'
+    compileSdkVersion 28
+    buildToolsVersion '28.0.3'
 
     defaultConfig {
         minSdkVersion 17
-        targetSdkVersion 23
+        targetSdkVersion 28
         versionCode 1
         versionName "1.0"
         ndk {
@@ -23,9 +23,9 @@ android {
 }

 dependencies {
-    provided fileTree(dir: 'libs', include: ['*.jar'])
-    compile "com.facebook.react:react-native:+"
+    compileOnly fileTree(dir: 'libs', include: ['*.jar'])
+    implementation "com.facebook.react:react-native:+"
     //compile 'com.yyl.vlc:vlc-android-sdk:3.0.13'
-    compile 'org.videolan:vlc-lib-android:3.1.7'
+    implementation 'org.videolan:vlc-lib-android:3.1.7'
     //compile 'de.mrmaffen:vlc-android-sdk:2.0.6'
 }
diff --git a/node_modules/react-native-vlcplayer2/android/src/main/java/com/yuanzhou/vlc/vlcplayer/ReactVlcPlayerView.java b/node_modules/react-native-vlcplayer2/android/src/main/java/com/yuanzhou/vlc/vlcplayer/ReactVlcPlayerView.java
index b40dfda..ce40b97 100644
--- a/node_modules/react-native-vlcplayer2/android/src/main/java/com/yuanzhou/vlc/vlcplayer/ReactVlcPlayerView.java
+++ b/node_modules/react-native-vlcplayer2/android/src/main/java/com/yuanzhou/vlc/vlcplayer/ReactVlcPlayerView.java
@@ -257,12 +257,12 @@ class ReactVlcPlayerView extends TextureView implements
 		*/
             DisplayMetrics dm = getResources().getDisplayMetrics();
             Media m = null;
-            if(netStrTag){
-                Uri uri = Uri.parse(this.src);
-                m = new Media(libvlc, uri);
-            }else{
-                m = new Media(libvlc, this.src);
-            }
+            // if(netStrTag){
+            Uri uri = Uri.parse(this.src);
+            m = new Media(libvlc, uri);
+            // }else{
+            //    m = new Media(libvlc, this.src);
+            // }
             m.setEventListener(mMediaListener);

             m.addOption(":rtsp-tcp");
diff --git a/node_modules/react-native-vlcplayer2/ios/RCTVLCPlayer/RCTVLCPlayer.m b/node_modules/react-native-vlcplayer2/ios/RCTVLCPlayer/RCTVLCPlayer.m
index eaeb530..30720e7 100644
--- a/node_modules/react-native-vlcplayer2/ios/RCTVLCPlayer/RCTVLCPlayer.m
+++ b/node_modules/react-native-vlcplayer2/ios/RCTVLCPlayer/RCTVLCPlayer.m
@@ -239,7 +239,7 @@ static NSString *const playbackRate = @"rate";
         int remainingTime = [[_player remainingTime] intValue];
         int duration      = [_player.media.length intValue];
         
-        if( currentTime >= 0 && currentTime < duration) {
+        if( currentTime >= 0 ) {
             self.onVideoProgress(@{
                                    @"target": self.reactTag,
                                    @"currentTime": [NSNumber numberWithInt:currentTime],
